capsules/src/process_console.rs

```diff
@@ -286,6 +243,61 @@ pub struct ProcessConsole<
     capability: C,
 }
 
+#[derive(Copy, Clone)]
+pub struct Command {
+    buf: [u8; COMMAND_BUF_LEN],
+    len: usize,
+}
+
+impl Command {
+    pub fn new() -> Command {
+        Command {
+            buf: [0; COMMAND_BUF_LEN],
+            len: 0,
+        }
+    }
+
+    /// Fill the buffer with the provided data.
+    /// If the provided data's length is smaller than the buffer length,
+    /// the left over bytes are not modified due to '\0' termination.
+    pub fn fill(&mut self, buf: &[u8; COMMAND_BUF_LEN], terminator_idx: usize) {
+        self.len = if terminator_idx >= COMMAND_BUF_LEN {
+            COMMAND_BUF_LEN
+        } else {
+            terminator_idx
+        };
+
+        (&mut self.buf).copy_from_slice(buf);
+    }
+
+    pub fn is_buffer_empty(&mut self) -> bool {
+        self.len == 0
+    }
+
+    pub fn get_byte(&self, byte_idx: usize) -> Option<u8> {
+        if byte_idx >= self.len {
+            return None;
+        }
+
+        Some(self.buf[byte_idx])
+    }
+}
+
+impl Default for Command {
+    fn default() -> Self {
+        Command {
+            buf: [0; COMMAND_BUF_LEN],
+            len: 0,
+        }
+    }
+}
+
+impl PartialEq<[u8; COMMAND_BUF_LEN]> for Command {
+    fn eq(&self, other_buf: &[u8; COMMAND_BUF_LEN]) -> bool {
+        self.buf.iter().zip(other_buf.iter()).all(|(a, b)| a == b)
```

Comment:
This should take into consideration only the bytes up to the terminator, no?

Reply:
I tried using try_while before the .all method and checked for both of them not to be equal with '\0', however this didn't work and did not add anything to the history